typing/includecore.ml

```diff
@@ -372,11 +387,28 @@ and compare_records ~loc env params1 params2 n
 let compare_records_with_representation ~loc env params1 params2 n
       labels1 labels2 rep1 rep2
   =
-  match compare_records ~loc env params1 params2 n labels1 labels2 with
-  | None when rep1 <> rep2 ->
-      let pos = if rep2 = Record_float then Second else First in
-      Some (Unboxed_float_representation pos)
-  | err -> err
+  let err = compare_records ~loc env params1 params2 n labels1 labels2 in
+  match err, rep1, rep2 with
+  | None, Record_regular, Record_regular
+  | None, Record_float, Record_float
+  | None, Record_unboxed _, Record_unboxed _
+  | None, Record_inlined _, Record_inlined _
+  | None, Record_extension _, Record_extension _ ->
+     None
+  | Some err, _, _ ->
+     Some (Record_mismatch err)
+  | None, Record_unboxed _, _ ->
+     Some (Unboxed_representation First)
```

Comment:
I apologies for the nitpick (plus side: I'm looking at this PR again!), but I find this pattern-match hard to read. At this point you know that the `_` in the third component is a representation distinct from `Record_unboxed`, but this is not so clear in the code. Here would be a concrete proposal (that could also be applied in `compare_variants`, even though it is shorter so more readable):

- deal with the `Some err` case first
- then group all the cases that deal with `Record_unboxed` together (success or failure), and then all the cases about `Record_float` together. (You can use an empty line between groups of cases.)

Reply:
I think this did make it clearer. Splitting it into two separate matches also helped.